### PR TITLE
Add penance and update textures

### PR DESCRIPTION
**Pull request by SekoiaTree**
As discussed in the discord.
Penance I blocks the curing process (a callback has been added for this, but curing still needs to be implemented)
Penance II kicks you out of your body and makes you a spirit (if you're a demon), and stops you from possessing shells. It does a bit of damage to mortals
Penance III stops you from possessing any bodies at all.

### CODE REVIEW

File: expansions/pandemonium/src/main/java/ladysnake/pandemonium/PandemoniumRequiemPlugin.java [hunk 1]
```diff
@@ -75,10 +82,20 @@ public void onRequiemInitialize() {
...
+                return TriState.FALSE;
+            }
+            return TriState.DEFAULT;
+        });
 
         // Enderman specific behaviour is unneeded now that players can possess them
         PossessionStartCallback.EVENT.unregister(new Identifier(Requiem.MOD_ID, "enderman"));
         PossessionStartCallback.EVENT.register(Pandemonium.id("allow_everything"), (target, possessor, simulate) -> PossessionStartCallback.Result.ALLOW);
+        PossessionStartCallback.EVENT.register(Pandemonium.id("deny_penance_two"), PenanceStatusEffect::canPossess);
+        //noinspection ConstantConditions
```
Comment (Pyrofab) [inline expansions/pandemonium/src/main/java/ladysnake/pandemonium/PandemoniumRequiemPlugin.java:96]: Could you add a comment to note why this suppression is necessary ?

File: expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                c.stopPossessing();
+            }
+            if (entity instanceof ServerPlayerEntity && !(RemnantComponent.get((PlayerEntity) entity)).isVagrant()) {
+                if (RemnantComponent.get((PlayerEntity) entity).getRemnantType().isDemon()) {
+                    PlayerSplitter.split((ServerPlayerEntity) entity);
+                } else {
+                    entity.damage(DamageSource.MAGIC, amplifier*4);
+                }
+            }
+        }
```
Comment (Pyrofab) [inline expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java]:
```suggestion
        if (amplifier >= 1 && entity instanceof ServerPlayerEntity) { // level 2+
            ServerPlayerEntity player = (ServerPlayerEntity) entity;
            if (amplifier >= 2) { // level 3+
	            PossessionComponent.get(player).stopPossessing();
	        }
            RemnantComponent remnant = RemnantComponent.get(player);
            if (!remnant.isVagrant()) {
                if (remnant.getRemnantType().isDemon()) {
                    PlayerSplitter.split(player);
                } else {
                    player.damage(DamageSource.MAGIC, amplifier*4);
                }
            }
        }
```

File: expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    public static PossessionStartCallback.@NotNull Result canPossess(MobEntity target, PlayerEntity possessor, boolean simulate) {
+        if (possessor.hasStatusEffect(PandemoniumStatusEffects.PENANCE) && possessor.getStatusEffect(PandemoniumStatusEffects.PENANCE).getAmplifier() >= 1 && target instanceof PlayerShellEntity) {
```
Comment (Pyrofab) [inline expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java]:
```suggestion
    public static PossessionStartCallback.@NotNull Result canPossess(MobEntity target, PlayerEntity possessor, boolean simulate) {
        StatusEffectInstance penance = possessor.getStatusEffect(PandemoniumStatusEffects.PENANCE);
        if (penance != null && penance.getAmplifier() >= 3 && target instanceof PlayerShellEntity) {
```

**Review by Pyrofab — CHANGES_REQUESTED** [review]

File: expansions/pandemonium/src/main/java/ladysnake/pandemonium/PandemoniumRequiemPlugin.java (hunk 1 above)
Comment (SekoiaTree) [reply expansions/pandemonium/src/main/java/ladysnake/pandemonium/PandemoniumRequiemPlugin.java:96]: Done

File: expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java (hunk 2 above)
Comment (SekoiaTree) [reply expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java]: That's not correct: amplifiers are 0-based, so amplifier >= 2 would be levels higher than 3. Also, the stopping of possession is starting at level 2, not level 3.

Comment (Pyrofab) [reply expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java]: The stopping at possession is indeed at level 3, not 2, at least in the plans we discussed on discord.
However you're very correct for the first part, I should have read your comment more carefully.

**Review by SekoiaTree — COMMENTED** [review]
This PR is mine, idk why it decided to ask me to review

**Review by Pyrofab — APPROVED** [review]
Great work, thank you very much !

**Pull request merged**